sdks/java/io/amazon-web-services/src/test/java/org/apache/beam/sdk/io/aws/s3/S3FileSystemTest.java

```diff
@@ -64,6 +77,28 @@
 /** Test case for {@link S3FileSystem}. */
 @RunWith(JUnit4.class)
 public class S3FileSystemTest {
+  private S3Mock api;
+  private AmazonS3 client;
+
+  @Before
+  public void before() {
+    api = new S3Mock.Builder().withPort(8001).withInMemoryBackend().build();
```

Comment:
Can we change this please to get the port dynamically to avoid possible test flakes see [here](https://github.com/apache/beam/blob/7c41e0a915083bd3b1fe52c2a417fa38a00e6463/runners/spark/src/test/java/org/apache/beam/runners/spark/translation/streaming/utils/EmbeddedKafkaCluster.java#L271) for an example. And can we make the embedded instance (api) part of a `@BeforeClass` so it gets instantiated only once.

Reply:
Done